Add the new userlog service (#396)
owncloud/docs-ocis: modules/ROOT/pages/deployment/services/s-list/userlog.adoc

= Userlog Service Configuration
:toc: right

> :description: The Infinite Scale userlog service is a mediator between the eventhistory service and clients who want to be informed about user related events. It provides an API to retrieve those.

Reviewer: ```suggestion
:description: The Infinite Scale userlog service is a mediator between the eventhistory service and clients who want to be informed about user-related events. It provides an API to retrieve those.
```

:oc10-api-url: https://doc.owncloud.com/server/next/developer_manual/core/apis/ocs-notification-endpoint-v1.html#get-user-notifications

:ext_name: userlog

// remember to REMOVE the corresponding tab if a new ocis release will be published

:no_second_tab: true
:no_third_tab: true

== Introduction

{description} 

== Prerequisites

Running the `userlog` service without running the xref:{s-path}/eventhistory.adoc[eventhistory] service is not possible.

== Storing

The `userlog` service persists information via the configured store in `USERLOG_STORE_TYPE`. Possible stores are:

[width=100%,cols="15%,85%",options=header]
|===
| Store Type
| Description

| `mem`
| Basic in-memory store and the default.

| `ocmem`
| Advanced in-memory store allowing max size.

| `Redis`
| Stores data in a configured Redis cluster.

| `etcd`
| Stores data in a configured etcd cluster.

| `nats-js`
| Stores data using the key-value-store feature of https://docs.nats.io/nats-concepts/jetstream/key-value-store[NATS JetStream].

| `noop`
| Stores nothing. Useful for testing. Not recommended in productive environments.
|===

1. Note that in-memory stores are by nature not reboot persistent.

> 2. Though usually not necessary, a database name and a database table can be configured for event stores if the event store supports this. Generally not applicapable for stores of type `in-memory`. These settings are blank by default which means that the standard settings of the configured store applies.

Reviewer: ```suggestion
2. Though usually not necessary, a database name and a database table can be configured for event stores if the event store supports this. Generally, this does not apply to stores of type `in-memory`. These settings are blank by default which means that the standard settings of the configured store applies.
```

3. The `userlog` service can be scaled if not using `in-memory` stores and the stores are configured identically over all instances.

== Configuring Events

> For the time being, the configuration which user related events are of interest is hardcoded and cannot be changed.

Reviewer: ```suggestion
Currently, the configuration which user-related events are of interest is hard-coded and cannot be changed.
```
